core/src/main/java/nl/nn/adapterframework/jdbc/StoredProcedureQuerySender.java

```diff
@@ -143,11 +199,17 @@ protected PreparedStatement prepareQueryWithColumnsReturned(Connection con, Stri
 	@Override
 	protected Message executeOtherQuery(Connection connection, PreparedStatement statement, String query, String resultQuery, PreparedStatement resStmt, Message message, PipeLineSession session, ParameterList parameterList) throws SenderException {
 		Message result = super.executeOtherQuery(connection, statement, query, resultQuery, resStmt, message, session, parameterList);
-		if (outputParameterPositions.length == 0) {
+		if (outputParameterDefs.length == 0) {
 			return result;
 		}
 		try {
-			return getResult(new StoredProcedureResultWrapper((CallableStatement) statement, statement.getParameterMetaData(), outputParameterPositions));
+			ParameterMetaData parameterMetaData;
+			if (getDbmsSupport().canFetchStatementParameterMetaData()) {
+				parameterMetaData = statement.getParameterMetaData();
```

Comment:
Is het niet 'gewoon' zo dat wanneer hier null uit komt het niet ondersteund wordt en anders wel? De `canFetchStatementParameterMetaData` lijkt zo overbodig?

Reply:
Maar er komt geen `null` uit, er komt een ParameterMetaData uit die exceptions gooit als je probeert iets uit te vragen (`SqlFeatureNotSupported` of zoiets).
In plaats van een nieuwe feature-flag zou ik ook kunnen controleren op `getDbmsSupport().getDbmsName().equals("Oracle")` maar dat lijkt me ook niet erg netjes.

Reply:
> Is het niet 'gewoon' zo dat wanneer hier null uit komt het niet ondersteund wordt en anders wel? De `canFetchStatementParameterMetaData` lijkt zo overbodig?

Helaas heb ik deze nog steeds nodig want voor sommige databases is onze mapping van Parameter Types naar JDBC types niet goed genoeg, en zouden we meer ParameterTypes nodig hebben.